### DISCUSSION TITLE
How to disable default locale appearing in all urls with dynamic routing

### DISCUSSION THREAD

**Discussion by bobarros** [vercel/next.js#23428] [Help] [answered]
I have the current next js config:
```
module.exports = {
  i18n: {
    locales: ['en', 'es', 'fr', 'pt'],
    defaultLocale: 'en',
    localeDetection: false,
  },
  reactStrictMode: true,
  trailingSlash: true
};
```

I'm rendering some pages with a file [url].js. If I try to use a path like:

```
params: {
        url: article.url,
},
```

All my paths become:

`mydomain.com/en/[url]`

If I try to use url: [ article.locale, article.url ], they become:

```
mydomain.com/en/pt/[url]
mydomain.com/en/es/[url]
mydomain.com/en/fr/[url]
```

What I really want is:

```
mydomain.com/en/[url]
mydomain.com/pt/[url]
mydomain.com/es/[url]
mydomain.com/fr/[url]
```
What am I missing?

**Comment by bobarros** [comment, marked as answer]
I found the answer. I need to pass the locale at getStaticPaths, otherwise default takes over.
```
export function getAllRoutes() {
  return mockedSample.map((article) => {
    return {
      params: {
        url: article.url
      },
      locale: "en or whatever",
    };
  });
}

export async function getStaticPaths() {
  const paths = await getAllRoutes();
  return {
    paths,
    fallback: false,
  };
}
```

**Reply by jamezening** [reply]
@bobarros 
Thank you for sharing this.
How did you end up implementing your solution?

I tried putting an empty string into locale =>

```
locale: '',
```

but the build still adds the default lang. What setting can prevent the default lang from being added?

**Reply by bobarros** [reply]
Hey, @gyoon-dev. I think my problem was a little bit different than yours. 

I had the default language in every url with the target one right after. Like www.mydomain.com/en/es

I think you want to have a default language with nothing added to the result url. Something like:

www.yourdomain.com instead of www.yourdomain.com/defaultLanguage

I don't know a proper solution and this thread below has a lot of discussion about it:

https://github.com/vercel/next.js/discussions/18419

I wish you luck.
